bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/internal/socket/BroadlinkSocket.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.broadlink.internal.socket;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.InetAddress;
+import java.net.MulticastSocket;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.broadlink.internal.ModelMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Threaded socket implementation
+ *
+ * @author John Marshall/Cato Sognen - Initial contribution
+ */
+
+@NonNullByDefault
+public class BroadlinkSocket {
+    private static byte buffer[];
+    private static DatagramPacket datagramPacket;
+    @Nullable
+    private static MulticastSocket socket = null;
+    @Nullable
+    private static Thread socketReceiveThread;
+    private static List<BroadlinkSocketListener> listeners = new ArrayList<BroadlinkSocketListener>();
+    private static final Logger LOGGER = LoggerFactory.getLogger(BroadlinkSocket.class);
```

Comment:
```suggestion
    private final Logger logger = LoggerFactory.getLogger(BroadlinkSocket.class);
```

As per guidelines...

Loggers should be non-static, final when ever possible and have the name logger.